File: src/main/java/org/lbogdanov/btlock/objc/Frameworks.java
package org.lbogdanov.btlock.objc;


import com.apple.jobjc.JObjC;
import com.apple.jobjc.JObjCRuntime;
import com.apple.jobjc.corebluetooth.CoreBluetoothFramework;
import com.apple.jobjc.corefoundation.CoreFoundationFramework;
import com.apple.jobjc.foundation.FoundationFramework;

public class Frameworks {
    public static final JObjCRuntime JOBJC;
    public static final FoundationFramework FOUNDATION;
    public static final CoreFoundationFramework CORE_FOUNDATION;
    public static final CoreBluetoothFramework CORE_BLUETOOTH;

    static {
        JObjC jobjc = JObjC.getInstance();
        FOUNDATION = jobjc.Foundation();
        CORE_FOUNDATION = jobjc.CoreFoundation();
        CORE_BLUETOOTH = jobjc.CoreBluetooth();
        JOBJC = JObjCRuntime.getInstance();

        JOBJC.registerUserClass(ManagerDelegate.class, ManagerDelegateClass.class);
    }
}

File: src/main/kotlin/btlock/objc/Frameworks.kt
package btlock.objc;

import com.apple.jobjc.*
import com.apple.jobjc.corebluetooth.*
import com.apple.jobjc.corefoundation.*
import com.apple.jobjc.foundation.*

object Frameworks {

    val CoreBluetooth:  CoreBluetoothFramework
    val CoreFoundation: CoreFoundationFramework
    val Foundation: FoundationFramework
    val objcNumbers: Utils.Numbers
    val objcRuntime: JObjCRuntime
    val objcStrings: Utils.Strings

    init {
        val jobjc = JObjC.getInstance()
        Foundation = jobjc.Foundation()
        CoreFoundation = jobjc.CoreFoundation()
        CoreBluetooth = jobjc.CoreBluetooth()
        objcRuntime = JObjCRuntime.getInstance()

        val utils = Utils.get()
        objcNumbers = utils.numbers()
        objcStrings = utils.strings()

        objcRuntime.registerUserClass(ManagerDelegate::class.java, ManagerDelegateClass::class.java)
    }

}
